### PR TITLE
Make machine_batch detect image_id like machine does

### PR DESCRIPTION
**Pull request by jkeiser**
This fails:

```ruby
machine_batch 'blah' do
  machine 'blah' do
  end
end
```

This succeeds:

```ruby
machine 'blah' do
end
```

Both should succeed.

### CODE REVIEW

**Comment by christinedraper** [comment]
Will this fix #89?

**Comment by jkeiser** [comment]
@christinedraper nope, I don't think so.  But that's merged now :)

File: lib/chef/provisioning/aws_driver/driver.rb [hunk 1]
```diff
@@ -61,6 +61,7 @@ def self.canonicalize_url(driver_url, config)
 
     # Load balancer methods
     def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
+      lb_options ||= {}
```
Comment (tyler-ball) [inline lib/chef/provisioning/aws_driver/driver.rb]: I prefer to see `lb_options = {}` in the method signature, but thats pretty personal.

**Comment by tyler-ball** [comment]
Other than minor style comment, :+1:

File: lib/chef/provisioning/aws_driver/driver.rb (hunk 1 above)
Comment (jkeiser) [inline lib/chef/provisioning/aws_driver/driver.rb]: Hmm, depends if machine_specs can ever not be passed.  I'll take a look at the caller and see if they can simply be persuaded to pass `{}` themselves.

Comment (jkeiser) [inline lib/chef/provisioning/aws_driver/driver.rb]: Note: I have submitted a PR to make it always pass machine_specs.

**Pull request closed without merging**